herbert_core/classes/instrument_classes/instrument/detectors/@IX_detector_array/IX_detector_array.m

```diff
@@ -146,8 +199,70 @@
         end
         
         %------------------------------------------------------------------
+        
+        function detpar = convert_to_old_detpar(obj)
+            detpar = struct();
+            if size(obj.det_bank.id,1)==1
+                detpar.group = obj.det_bank.id;
+                detpar.x2    = obj.det_bank.x2;
+                detpar.phi   = obj.det_bank.phi;
+                detpar.azim  = obj.det_bank.azim;
+                detpar.width = obj.det_bank.det.dia;
+                detpar.height = obj.det_bank.det.height;
+            else
+                detpar.group = obj.det_bank.id';
+                detpar.x2    = obj.det_bank.x2';
+                detpar.phi   = obj.det_bank.phi';
+                detpar.azim  = obj.det_bank.azim';
+                detpar.width = obj.det_bank.det.dia';
+                detpar.height = obj.det_bank.det.height';
+            end
+            detpar.filename = obj.filename;
+            detpar.filepath = obj.filepath;
+        end
+    end
+    
+    methods(Static)
+        function is_dp_struct = check_detpar_parms(dp)
```

Comment:
Does this name represent what the function is doing?

Reply:
it will do till after the merge